core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/EqualsExpression.kt

```diff
@@ -1,16 +1,48 @@
 package ch.ergon.dope.resolvable.expression.unaliased.type.relational
 
 import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.toBooleanType
+import ch.ergon.dope.resolvable.expression.unaliased.type.toNumberType
+import ch.ergon.dope.resolvable.expression.unaliased.type.toStringType
 import ch.ergon.dope.resolvable.operator.InfixOperator
 import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.NumberType
+import ch.ergon.dope.validtype.StringType
 import ch.ergon.dope.validtype.ValidType
 
-class EqualsExpression<T : ValidType>(
-    val left: TypeExpression<T>,
-    val right: TypeExpression<T>,
+class EqualsExpression<out T : ValidType>(
+    val left: TypeExpression<out T>,
```

Comment:
Do we really want _out_ here? Now we can write stuff like:
`EqualsExpression("5".toStringType(), 5.toNumberType())` and loose the type safety again. 
You are basically writing here: We can equal any two Valid-Types

Reply:
No we dont want that here. Removed out